src/pudl_archiver/archivers/classes.py

```diff
@@ -364,6 +367,73 @@ def _check_dataset_size(
             required_for_run_success=self.fail_on_dataset_size_change,
         )
 
+    def _check_data_continuity(
+        self, new_datapackage: DataPackage
+    ) -> validate.DatasetSpecificValidation:
+        """Check that the archived data partitions are continuous and unique."""
+        success = True
+        note = None
+        partition_names = []
+        dataset_partitions = []
+
+        # Unpack partitions of a dataset.
+        for resource in new_datapackage.resources:
+            if not resource.parts:  # Skip resources without partitions
+                continue
+            partition_names += list(resource.parts)
+            dataset_partitions += list(resource.parts.values())[0]
+
+        partition_names = set(partition_names)
+
+        # Only perform this test if the part label is year quarter or year month
+        # Note that this currently only works if there is one set of partitions,
+        # and will fail if year_month and form are used to partition a dataset, e.g.
+        if any(partition in VALID_PARTITION_RANGES for partition in partition_names):
+            partition_to_test = [
+                partition
+                for partition in partition_names
+                if partition in VALID_PARTITION_RANGES
+            ][0]
```

Comment:
this isn't the case rn but if an archive has multiple partitions in `VALID_PARTITION_RANGES` we'll only test the first one we find

Reply:
I originally had it designed with this in mind but I talked to @e-belfer and she said that this would probably never happen.

Reply:
I can't imagine us wanting to partition both by `year_quarter` and `year_month`, and this raises a whole other host of questions - are we enforcing consecutiveness within all the partition values of one key for one value of the other key (e.g., all the `year_months` by `form`, if `form` were capable of being consecutive?). I agree the silent failure is bad and have explicitly added a test failure to flag this.